### PR TITLE
drivers: tmag5273: remove unneccessary check

### PR DESCRIPTION
**Pull request by deveritec-jusc**
If an immediate return on error is used, an additional result check in `sample_fetch` can be neglected. This doesn't change the behavior, as for the fetch routine additionally checked for an unset `TMAG5273_DIAG_FAIL`.

### CODE REVIEW

File: drivers/sensor/ti/tmag5273/tmag5273.c [hunk 1]
```diff
@@ -144,23 +144,24 @@ static int tmag5273_check_device_status(const struct tmag5273_config *drv_cfg,
...
+		LOG_ERR("INT pin error detected");
 	}
 
 	if ((*device_status & TMAG5273_OSC_ER_MSK) == TMAG5273_OSC_ERR) {
-		LOG_WRN("Oscillator error detected");
+		LOG_ERR("Oscillator error detected");
 	}
 
-	return 0;
+	LOG_ERR("unknown error");
```
Comment (ubieda) [inline drivers/sensor/ti/tmag5273/tmag5273.c:163]: We'd hit this error even when the error is known.
```suggestion
```

**Review by ubieda — COMMENTED** [review]
The removal of the additional check looks good at a glance. One nit captured before I give a +1

File: drivers/sensor/ti/tmag5273/tmag5273.c (hunk 1 above)
Comment (deveritec-rosc) [reply drivers/sensor/ti/tmag5273/tmag5273.c]: good catch! removed it.

**Review by ubieda — APPROVED** [review]

File: drivers/sensor/ti/tmag5273/tmag5273.c [hunk 2]
```diff
@@ -102,17 +102,17 @@ static int tmag5273_reset_device_status(const struct device *dev)
 /**
  * @brief checks for DIAG_FAIL errors and reads out the DEVICE_STATUS register if necessary
  *
- * Prints a human readable representation to the log, if \c CONFIG_LOG is activated.
- *
- * @param drv_cfg[in] driver instance configuration
- * @param device_status[out] DEVICE_STATUS register if DIAG_FAIL is set
+ * @param[in] drv_cfg driver instance configuration
+ * @param[out] device_status DEVICE_STATUS register if DIAG_FAIL is set
```
Comment (MaureenHelm) [inline drivers/sensor/ti/tmag5273/tmag5273.c:106]: This and some other changes appear unrelated to the commit message `drivers: tmag5273: remove unneccessary check`. Please revert the unrelated changes or move them to separate commits.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/ti/tmag5273/tmag5273.c (hunk 2 above)
Comment (deveritec-rosc) [reply drivers/sensor/ti/tmag5273/tmag5273.c:106]: Better now?

**Review by MaureenHelm — APPROVED** [review]
